poetry/repositories/legacy_repository.py

```diff
@@ -395,4 +395,4 @@ def _get(self, endpoint):  # type: (str) -> Union[Page, None]
             )
             return
 
-        return Page(url, response.content, response.headers)
+        return Page(response.url, response.content, response.headers)
```

Comment:
I think you should also print `response.url` in the log message on line 394 to make it easier to debug.  `requests` (i.e. the call to `self.session.get()`) may return the error from a different url then the one initially requested because `requests` can automatically follow redirects.  I believe it would be easier to debug an issue if the final url is known rather than the initially requested url.